### PR TITLE
Fix serialize_work and restore_work to take video and restricted derivative storage into account.

### PR DESCRIPTION
**Pull request by eddierubeiz**
Ref #1600 .
Spent some time cleaning up restore_work. 
Restricted derivatives weren't working; fixed them as well.

Try e.g.
```
# restricted derivatives
bundle exec rake scihist:heroku:copy_data[bwv8gvz,true]
# video test
bundle exec rake scihist:heroku:copy_data[bhfkgwq, true]
```
To test this code, both dev and staging need to be running it.

### CODE REVIEW

File: app/models/copy_staging/restore_work.rb [hunk 1]
```diff
@@ -32,27 +32,24 @@ module CopyStaging
...
       @json_file = json_file
       @thread_pool_size = thread_pool_size
-
       @thread_pool = Concurrent::FixedThreadPool.new(thread_pool_size)
       @tracked_futures = Concurrent::Array.new
-
-      # make sure they get registered
-      remote_store_storage
-      remote_derivatives_storage
+      @storage_map = Hash[ (ORIGINALS_STORAGE + DERIVATIVES_STORAGE).collect { |v| [v, "remote_#{v.to_s}".to_sym  ] } ]
```
Comment (jrochkind) [inline app/models/copy_staging/restore_work.rb:51]: I honestly would have just written this out the long way, for clarity... but no big deal, not requiring the change, just saying.

File: app/models/copy_staging/serialize_work.rb [hunk 2]
```diff
@@ -85,5 +82,14 @@ def shrine_config(shrine_storage_key)
         "prefix" => storage.prefix
       }
     end
+
+    # The list of Shrine storages we can copy from and to
+    # is listed as a constant in
+    # CopyStaging::RestoreWok, so let's use that here.
+    def storages
+      storage_list = CopyStaging::RestoreWork::ORIGINALS_STORAGE + CopyStaging::RestoreWork::DERIVATIVES_STORAGE
+      Hash[ storage_list.collect { |v| [v.to_s, shrine_config(v)  ] } ]
```
Comment (jrochkind) [inline app/models/copy_staging/serialize_work.rb:91]: Again all this indirection is seeming. bit confusing ot me, I honestly would have just written it out here, even if it duplicates things that are other places. Becuase then the develoepr could see exactly what's going on, right now... if I come back to this and want to know what's going on, it's going to take me a while to go back and forth between the multiple data structures in multiple files and figure out what the heck is going on.

**Comment by jrochkind** [comment]
I'm finding the indirection/abstraction a bit confusing... but maybe it's necessarily going to be confusing just becuase of what's going on? I'm not sure. What do you think, if you came to this not having written it, would you find it easy to see what's going on? Is there anything that can be done to make it easier? Maybe more comments, but maybe just simplifying the code with less abstraction/indirection, just spelling out the storage details where they are used?

But I leave it up to your judgement, approved to merge when you think it's ready. 

Thanks for working on this kind of hacky stuff!

File: app/models/copy_staging/serialize_work.rb (hunk 2 above)
Comment (eddierubeiz) [reply app/models/copy_staging/serialize_work.rb]: Good call -- -added some more documentation to clear this up a bit.

**Pull request merged**